src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -2741,14 +2744,15 @@ void VM_Version::crac_restore() {
   // Workaround JDK-8311164: CPU_HT is set randomly on hybrid CPUs like Alder Lake.
   features_missing &= ~CPU_HT;
 
-  if (features_missing || glibc_features_missing) {
+  _crac_restore_missing_features = features_missing || glibc_features_missing;
+  if (_crac_restore_missing_features) {
     static const char part1[] = "You have to specify -XX:CPUFeatures=";
     tty->print_raw(part1, sizeof(part1) - 1);
     nonlibc_tty_print_uint64_comma_uint64(_features & features_saved, _glibc_features & glibc_features_saved);
     static const char part2[] = " together with -XX:CRaCCheckpointTo when making a checkpoint file; specified -XX:CRaCRestoreFrom file contains CPU features ";
     tty->print_raw(part2, sizeof(part2) - 1);
     nonlibc_tty_print_uint64_comma_uint64(features_saved, glibc_features_saved);
-    fatal_missing_features(features_missing, glibc_features_missing);
+    missing_features(features_missing, glibc_features_missing);
   }
 
   auto supports_exit = [&](const char *supports, bool file, bool this_cpu) {
```

Comment:
Just a note, not a part of this PR: all `_supports_*`, except _supports_cx8, are hard-wired to `true`, and all of them mostly serve to communicate CPU capabilities to the shared code.

The `_supports_cx8` equals to supports_cmpxchg8(), and I assume this is supported pretty widely. And supports_cmpxchg8 implemented via` _features & CPU_CX8`, so that is tested along CPU features check.

It looks the whole handling of `_supports_*` can be removed, or at least be turned into assert()s.

Reply:
The `_supports_*` code It has been removed by #103 so it is no longer relevant. `CX8` is not supported by 32-bit CPUs.